File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice4.c
#include <stdio.h>
int NotesS_M,C_max;
float max,somme,moyenne,T[40];
int main(){
    somme=0;
    for(int i=0;i<40;i++){
        printf("saisir la note %d ",i+1);
        scanf("%f",&T[i]);
        somme+=T[i];
    }
    moyenne=somme/40;
    max=T[0];
    for(int i=0;i<40;i++){
        if(T[i]>max)
            max=T[i];
        if(T[i]>moyenne)
            NotesS_M++;
    }
    for(int i=0;i<40;i++){
        if(T[i]==max)
            C_max++;
    }
    printf("le nombre des notes supérieures strictement à la moyenne %f  est %d\n",moyenne,NotesS_M);
    printf("le nombre d’occurrence de la note maximale est %d\n",C_max);
    return 0;
}

File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_4/exercice4.py
T=[None for i in range(40)]
somme=0
for i in range(0,40):
    T[i]=float(input(f"saisir la note {i+1}: "))
    somme+=T[i]
max=T[0]
moyenne=somme/40
Notes_SS_M=0
for i in range(0,40):
    if T[i]>max :
        max=T[i]
    if T[i]>moyenne :
        Notes_SS_M+=1
C_max=0
for i in range(0,40):
    if T[i]==max:
        C_max+=1
print(f"le nombre des notes supérieures strictement à la moyenne {moyenne} est {Notes_SS_M}.")
print("le nombre d’occurrence de la note maximale est ",C_max)